language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/BallerinaCompletionContextImpl.java

```diff
@@ -82,4 +86,25 @@ public void addResolver(Node node) {
     public List<Node> getResolverChain() {
         return this.resolverChain;
     }
+
+    @Override
+    public Optional<TypeSymbol> getContextType() {
+        if (this.contextType != null) {
+            return this.contextType;
+        }
+        if (getNodeAtCursor() == null) {
+            return Optional.empty();
+        }
+        ContextTypeResolver contextTypeResolver = new ContextTypeResolver(this);
+        this.contextType = getNodeAtCursor().apply(contextTypeResolver);
+
+        NonTerminalNode node = getNodeAtCursor();
+        while (contextType == null && node.parent() != null) {
+            this.contextType = Optional.ofNullable(node.parent().apply(contextTypeResolver).get());
+        }
+        if (contextType == null) {
+            contextType = Optional.empty();
```

Comment:
Revamp this logic according to the above comment

Reply:
Done